lib/Sema/CSDiag.cpp

```diff
@@ -4264,8 +4264,14 @@ bool TypeChecker::diagnoseSelfAssignment(const Expr *E) {
   if (!AE)
     return false;
 
+  // Allow the use of parentheses to express self-assignment when it's a deliberate choice
+  auto src = AE->getSrc();
+  auto loadSrc = dyn_cast<LoadExpr>(src);
```

Comment:
Thanks for taking the time to give such a detailed explanation, @rudkx. I'll look over that information and revise accordingly.